### DISCUSSION TITLE
[Feature]:  Gallery App Functionality

### DISCUSSION THREAD

**Discussion by anonymoushand** [immich-app/immich#1605] [Feature Request]
### Feature detail

code immich as a gallery app as well. I.E. currently if I download a picture on android, and open it I would have to open it in google photos as Immich isn't shown as an option through that context menu or able to be set as default image viewer. This would be big for making Immich usable as the only needed image app on android.

As explained from Jrasm91 in discord:"I think there are a few different levels to that, but you're basically talking about registering immich to handle other "intents", such as image mime-types. So when you open an image, immich is in the list of available app, right? And then you'd have the option to set immich as the default handler as well"

### Platform

Mobile App

**Comment by 120318** [comment]
any progress？

**Comment by fskaeh** [comment]
+1 on this, it would be a great convenience feature.

**Comment by SlavikCA** [comment]
> @alextran1502  We just add a permission in Android for this. Can you help double check in Immich app setting on Android?

I have Android app 1.99.0 and can't use Immich as "photo viewer app".
Tested with apps:
- Camera
- Google Voice

Few screenshots here:
https://github.com/immich-app/immich/discussions/8116

**Reply by alextran1502** [reply]
Ah there seems more to be done to get to that point

**Comment by AZweimiller** [comment]
IMO there's a few things that are needed to make the Immich mobile app more competitive as a drop-in gallery app. These items are mentioned across a few different discussions but I thought it would help to summarize them in one place. This discussion felt appropriate due to the title. I am using Android so these items may not apply to iOS.

1. Register as a viewer for media opened from other apps.
    https://github.com/immich-app/immich/discussions/1605 (This discussion)
    https://github.com/immich-app/immich/discussions/7612
    https://github.com/immich-app/immich/discussions/7363

2. Ability to browse all albums and media on device, not just albums that are selected for auto-backup. This could be expanded to allow manual uploading of assets in these albums.
    https://github.com/immich-app/immich/discussions/7075

3. Automatically delete assets from mobile devices when removed from web.
    https://github.com/immich-app/immich/discussions/3594

These are all features of the `Other App`.

Thank you.

**Reply by alextran1502** [reply]
We will get there!

**Reply by benmccann** [reply]
For point 1, https://github.com/immich-app/immich/discussions/7612 seems to actually be asking for the reverse. I.e. the ability to open a photo from Immich into Google Photos to allow for editing it

**Reply by alecdotdev** [reply]
> the ability to open a photo from Immich into Google Photos to allow for editing it

Exactly

**Reply by benmccann** [reply]
The discussions for item 2 are a bit hard to follow, but I believe that item is now implemented: https://github.com/immich-app/immich/discussions/7075#discussioncomment-13974023

**Reply by benmccann** [reply]
It looks like there's a related feature of sharing from Immich: https://github.com/immich-app/immich/pull/23155

**Comment by krzemyczek** [comment]
This could be so great to convince my whole family into using immich and ditching google photos

**Comment by xxxsskxxx** [comment]
Hi all, just checking, is this feature planned? WOuld love to get rid of google photos and replace it completely with Immich! Thanks

**Reply by krzemyczek** [reply]
As for now, i use https://f-droid.org/packages/com.dot.gallery/. Which is the best open source gallery app which I found

**Reply by JordyEGNL** [reply]
> As for now, i use https://f-droid.org/packages/com.dot.gallery/. Which is the best open source gallery app which I found

Thanks I will use that instead of Google Photos for now!

**Comment by Serinus1** [comment]
I've gotten rid of Google photos, and just use Files for anything non-Immich (like screenshots).  The biggest annoyance is not being able to open photos directly from the default Camera app.

**Reply by krzemyczek** [reply]
Which camera app you use? I'm afraid that many of them doesn't allow 3rd party ones

**Comment by scan5415** [comment]
Hi all
Do you have any update for this android features?

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Reply by KweezyCode** [reply]
> https://justinmayer.com/posts/any-updates/

It's very strange to send such links, having 500 open issues. It often happens that developers forget to close the problem, having implemented the functionality long ago. I think you could spend your time more effectively by simply answering "yes" or "no" than looking for this link

**Reply by bo0tzz** [reply]
@KweezyCode every unnecessary comment on a thread notifies dozens of people, and there's a handful of them each day - which also means time spent by us replying instead of doing more useful things with our time. "Looking for" this link was instant as I have it in my bookmarks. I post it because I'm hoping to reduce the amount of comments like this, for the reasons above.

**Comment by basicuser01** [comment]
Hello, I am really wishing this feature was active. I want to be able to set Immich as my default photos app so that I can select photos from it from other applications. Thank you!

**Comment by chancedonmillion** [comment]
I too would like this feature.

**Comment by serj-47** [comment]
Is it a pity that there is no movement in this direction? the option is extremely necessary. For example, in my iqoo phone, I can select the default gallery app, but it is impossible to select immich in this list.

**Comment by denysapplications** [comment]
Same issue on OnePlus, after disabling Google Photos, there is no app that can open the last picture from the camera app.
Opening Immich this way would be great.
I know it is possible, because alternative gallery apps such as Fossify gallery do the trick. After installing it opens that gallery from the camera app.

**Comment by BerziOnline** [comment]
+1

**Comment by jx8g2** [comment]
Any updates on this feature? Seems more relevant now with the free up space feature on immich as more photos are being accessed remotely.

**Comment by pkoevesdi** [comment]
Same here on a Fairphone 5, and desperately needed. Also have it working with several other gallery apps from F-Droid, so it's definately possible.

**Comment by bo0tzz** [comment]
Folks, there are many people subscribed to this thread and every unnecessary comment notifies all of them. If you just want to express your support for this feature request, you should do that by using the upvote button on the main post.

**Comment by benmccann** [comment]
I've sponsored Peter to work on this. PR incoming soon!

**Reply by benmccann** [reply]
PR here: https://github.com/immich-app/immich/pull/26109

**Comment by mnagel** [comment]
Thank you very much to everyone involved!

I'm running Immich App 3.0.1 (build 3054). One small suggestion: it would be very useful if the version information on the About screen could be copied, as it currently has to be typed manually.

The original issue is resolved for me in apps such as Material Files. I can now select Immich to open JPG files there, which works exactly as expected.
However, I still see an inconsistency with Open Camera. When choosing an app to view a captured photo, Open Camera only offers Google Photos and the Samsung Gallery app. Immich is not available as an option.
Would it be possible to extend the fix to cover this use case as well? I think Immich and Open Camera are a great combination for users looking for an open, privacy-friendly photo workflow.

I'm happy to provide any additional information or run tests if that would help.

Thanks,
Michael
